platform/windows/os_windows.cpp

```diff
@@ -1850,16 +1850,35 @@ String OS_Windows::get_cache_path() const {
 		if (has_environment("LOCALAPPDATA")) {
 			cache_path_cache = get_environment("LOCALAPPDATA").replace("\\", "/");
 		}
-		if (cache_path_cache.is_empty() && has_environment("TEMP")) {
-			cache_path_cache = get_environment("TEMP").replace("\\", "/");
-		}
 		if (cache_path_cache.is_empty()) {
-			cache_path_cache = get_config_path();
+			cache_path_cache = get_tmp_path();
 		}
 	}
 	return cache_path_cache;
 }
 
+String OS_Windows::get_tmp_path() const {
+	static String tmp_path_cache;
+	if (tmp_path_cache.is_empty()) {
+		{
+			// Get the tmp path from the Windows API.
+			void *get_temp_path = (void *)GetProcAddress(GetModuleHandleA("kernel32.dll"), "GetTempPathW");
+			if (get_temp_path) {
+				Vector<WCHAR> tmp_path;
+				tmp_path.resize(MAX_PATH);
+				DWORD tmp_path_length = GetTempPathW(MAX_PATH, (LPWSTR)tmp_path.ptrw());
+				if (tmp_path_length > 0 && tmp_path_length < MAX_PATH) {
+					tmp_path_cache = String::utf16((const char16_t *)tmp_path.ptr(), tmp_path_length / sizeof(WCHAR));
```

Comment:
```suggestion
					tmp_path_cache = String::utf16((const char16_t *)tmp_path.ptr(), tmp_path_length);
```
or
```suggestion
					tmp_path_cache = String::utf16((const char16_t *)tmp_path.ptr());
```

It's alreay number of characters, and that's what `utf16` is excpecting, but since string should be null terminated it should work without the size.